kimchi/src/circuits/gate.rs

```diff
@@ -68,40 +69,67 @@ pub struct SingleLookup<F> {
 /// analogously using `joint_combiner`.
 ///
 /// This function computes that combined value.
-pub fn combine_table_entry<'a, F: Field, I: DoubleEndedIterator<Item = &'a F>>(
+pub fn combine_table_entry<
+    'a,
+    F: 'a + Zero + One + Add + Mul + Clone,
+    I: DoubleEndedIterator<Item = &'a F>,
+>(
     joint_combiner: F,
     v: I,
 ) -> F {
-    v.rev().fold(F::zero(), |acc, x| joint_combiner * acc + x)
+    v.rev()
+        .fold(F::zero(), |acc, x| joint_combiner.clone() * acc + x.clone())
 }
 
-impl<F: Field> SingleLookup<F> {
+impl<F: Copy> SingleLookup<F> {
```

Comment:
is there a reason behind that relaxation and the creation of K? Why not just take everything as a `Field`?

Reply:
`K` can be `Expr`; that was the point of this PR.

Reply:
*facepalm* I did get straight into the code without reading the description